src/battle_script_commands.c

```diff
@@ -9957,22 +9957,22 @@ static void Cmd_various(void)
         else
         {
             gSpecialStatuses[gBattlerTarget].instructedChosenTarget = *(gBattleStruct->moveTarget + gBattlerTarget) | 0x4;
-            gBattlerAttacker = gBattlerTarget;
+            gBattleStruct->savedBattlerAttacker[gBattleStruct->savedAttackerCount++] = gBattlerTarget;
```

Comment:
is it possible to stare attacker and target in the script before attackcancler?

Reply:
I have an alternative that uses gEffectBattler instead of saving and restoring attacker/target, as long as that doesn't cause any problems